pkg/resource/detector/detector.go

```diff
@@ -0,0 +1,101 @@
+package detector
+
+import (
+	runtime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"time"
+)
+
+// Detector represents a procedure that runs in the background, periodically auto-detecting features
+type Detector struct {
+	dc                  discovery.DiscoveryInterface
+	ticker              *time.Ticker
+	SubscriptionChannel chan schema.GroupVersionKind
+	crds 				map[runtime.Object]trigger
+}
+
+type trigger func(runtime.Object)
+
+// New creates a new auto-detect runner
+func NewAutoDetect(dc discovery.DiscoveryInterface) (*Detector, error) {
+	// Create a new channel that GVK type will be sent down
+	subChan := make(chan schema.GroupVersionKind, 1)
+
+	return &Detector{dc: dc, SubscriptionChannel: subChan, crds:map[runtime.Object]trigger{}}, nil
+}
+
+
+//AddCRDTrigger to run the trigger function,
+//the first time that the background scanner discovers that the CRD type specified exists
+func (d *Detector) AddCRDTrigger(crd runtime.Object, trigger trigger) {
+	d.crds[crd] = trigger
```

Comment:
@philbrookes Do you think there should be a check here to ensure the runtime object passed in is the specific CRD format specified. Otherwise we will get an exception from https://github.com/RHsyseng/operator-utils/pull/26/files#diff-878cddc74756d7c83223de72f5c0dbfeR73 when it doesn't have the method `GetObjectKind`

Reply:
if an object doesn't have `GetObjectKind` that it wouldn't satisfy the `runtime.Object` interface:
https://github.com/kubernetes/apimachinery/blob/master/pkg/runtime/interfaces.go#L299

Reply:
Excellent, good to know. Thank you.